bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/HydrawiseHandler.java

```diff
@@ -0,0 +1,321 @@
+            ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), channelName);
+            logger.debug("updateState updating {} {}", channelUID, state);
+            updateState(channelUID, state);
+        }
+    }
+
+    @SuppressWarnings("serial")
+    protected class NotConfiguredException extends Exception {
+        NotConfiguredException(String message) {
+            super(message);
+        }
+    }
+
+    private boolean isFutureValid(@Nullable ScheduledFuture<?> future) {
+        return future != null && !future.isCancelled();
+    }
+
+    private void configureInternal() {
+        clearPolling();
+        stateMap.clear();
+        relayMap.clear();
+        try {
+            configure();
+            initPolling(0);
+        } catch (NotConfiguredException e) {
+            logger.debug("Configuration error {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (HydrawiseConnectionException e) {
+            logger.debug("Could not connect to service");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (HydrawiseAuthenticationException e) {
+            logger.debug("Credentials not valid");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Credentials not valid");
+        }
+    }
+
+    /**
+     * Starts/Restarts polling with an initial delay. This allows changes in the poll cycle for when commands are sent
+     * and we need to poll sooner then the next refresh cycle.
+     */
+    private synchronized void initPolling(int initalDelay) {
+        clearPolling();
+        pollFuture = scheduler.scheduleWithFixedDelay(this::pollControllerInternal, initalDelay, refresh,
+                TimeUnit.SECONDS);
+    }
+
+    /**
+     * Stops/clears this thing's polling future
+     */
+    private void clearPolling() {
+        ScheduledFuture<?> localFuture = pollFuture;
+        if (isFutureValid(localFuture)) {
+            if (localFuture != null) {
+                localFuture.cancel(false);
+            }
+        }
+    }
+
+    /**
+     * Poll the controller for updates.
+     */
+    private void pollControllerInternal() {
+        try {
+            pollController();
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (HydrawiseConnectionException e) {
+            // poller will continue to run, set offline until next run
+            logger.debug("Exception polling", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (HydrawiseAuthenticationException e) {
+            // if are creds are not valid, we need to try re authorizing again
+            logger.debug("Authorization exception during polling", e);
+            clearPolling();
```

Comment:
Is this `clearPolling()` needed? `configureInternal()`  also calls `clearPolling()`.